src/Types/CborBytes.purs

```diff
@@ -0,0 +1,79 @@
+-- | CborBytes. A wrapper over `ByteArray` to indicate that the bytes are cbor.
+
+module Types.CborBytes
+  ( CborBytes(..)
+  , cborBytesToByteArray
+  , cborBytesFromByteArray
+  , cborBytesToIntArray
+  , cborBytesFromIntArray
+  , cborBytesFromIntArrayUnsafe
+  , cborBytesToHex
+  , byteLength
+  , hexToCborBytes
+  , hexToCborBytesUnsafe
+  ) where
+
+import Data.Newtype (class Newtype, wrap, unwrap)
+
+import Types.ByteArray (ByteArray)
+import Types.ByteArray as BA
+import Data.Maybe (Maybe)
+import Prelude
+import Aeson (class DecodeAeson, decodeAesonViaJson)
+import Data.Argonaut (class DecodeJson, decodeJson)
+import Test.QuickCheck.Arbitrary (class Arbitrary, arbitrary)
+
+-- | An array of Bytes containing CBOR data
+newtype CborBytes = CborBytes ByteArray
+
+instance Show CborBytes where
+  show rb = "(cborBytesFromIntArrayUnsafe" <> show (cborBytesToIntArray rb) <>
+    ")"
+
+derive instance Newtype CborBytes _
+
+derive instance Eq CborBytes
+
+derive instance Ord CborBytes
+
+instance Semigroup CborBytes where
+  append r1 r2 = wrap $ unwrap r1 `append` unwrap r2
+
+instance Monoid CborBytes where
+  mempty = cborBytesFromIntArrayUnsafe []
```

Comment:
I would write this as `mempty = wrap mempty` but down to you. Although see the comment below.